core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -225,6 +226,68 @@ public int hashCode() {
 
 // norelease: we need to override freeze() and add safety checks that all settings are actually set
 
+    /**
+     * Checks for any conflicts between this field type and other.
+     */
+    public void validateCompatible(MappedFieldType other, List<String> conflicts) {
+        boolean indexed =  indexOptions() != IndexOptions.NONE;
+        boolean mergeWithIndexed = other.indexOptions() != IndexOptions.NONE;
+        if (indexed != mergeWithIndexed || tokenized() != other.tokenized()) {
```

Comment:
Should we validate index options as well? ie. to prevent adding offsets while only freqs were indexed until now?